rlbot_gui/match_runner/match_runner.py

```diff
@@ -118,6 +118,17 @@ def fetch_game_tick_packet() -> GameTickPacket:
     return game_tick_packet
 
 
+def get_fresh_setup_manager():
+    global sm
+    if sm is not None:
+        try:
+            sm.shut_down()
+        except Exception as e:
+            print(e)
+    sm = SetupManager()
+    return sm
```

Comment:
Do you mind on elaborating on why sharing the SetupManager helps?

Reply:
Allows this loop to hit the active setup manager instance: https://github.com/RLBot/RLBotGUI/blob/master/rlbot_gui/gui.py#L621-L623

Which does this https://github.com/RLBot/RLBotGUI/blob/67e86702fc06b4b34809d09231071f82ba247312/rlbot_gui/match_runner/match_runner.py#L178-L179, which facilitates hive mind assembly :)